### DISCUSSION TITLE
CSRF name changed in v2.x

### DISCUSSION THREAD

**Discussion by rubanraj-r** [ory/hydra#3801] [Q&A] [answered]
with version 2.x, I can see some random number suffixed to the login csrf name and also the default csrf name is changed. Why is this change?

**Comment by terev** [comment, marked as answer]
The random number is actually a hash of the oauth client's id.  Seems to be so that login session csrf tokens are isolated based on the oauth client. In turn this allows concurrent login sessions for different oauth clients to succeed.

**Comment by rubanraj-r** [comment]
Thanks @terev
